hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -225,9 +223,18 @@ func (r *NodePoolReconciler) reconcile(ctx context.Context, hcluster *hyperv1.Ho
 	switch nodePool.Spec.Platform.Type {
 	case hyperv1.AWSPlatform:
 		md := machineDeployment(nodePool, hcluster.Spec.InfraID)
-		ami, err := r.ImageProvider.Image(hcluster)
-		if err != nil {
-			return ctrl.Result{}, fmt.Errorf("failed to obtain AMI: %w", err)
+		var ami string
+		if arch, foundArch := releaseImage.StreamMetadata.Architectures["x86_64"]; !foundArch {
```

Comment:
Do you think this new logic deserves its own unit testable function?

Reply:
I do have some tests in progress I can add to the PR, yes

Reply:
I did extract some of the logic into a function, but it's so small it doesn't seem worth unit testing at the moment — I do think the reconciliation code itself deserves tests but that would be a much larger refactor (there aren't any tests at all for this controller currently, and I think it needs some restructuring to make that manageable), so I'm not tackling it in this PR